diff -Nur kvm-75.broken/qemu/dyngen.c kvm-75/qemu/dyngen.c
--- kvm-75.broken/qemu/dyngen.c	2008-09-07 18:38:33.000000000 +0200
+++ kvm-75/qemu/dyngen.c	2008-09-21 01:39:07.000000000 +0200
@@ -1347,8 +1347,9 @@
         p = (void *)(p_end - 4);
         if (p == p_start)
             error("empty code for %s", name);
-        if (get32((uint32_t *)p) != 0x4e800020)
-            error("blr expected at the end of %s", name);
+	uint32_t last_op = get32( (uint32_t *) p );
+        if ( (last_op != 0x4e800020) && (( last_op & 0xfc000003) != 0x48000000) )
+            error("blr or b expected at the end of %s, got: 0x%.8x", name, last_op );
         copy_size = p - p_start;
     }
 #elif defined(HOST_S390)
diff -Nur kvm-75.broken/qemu/Makefile.target kvm-75/qemu/Makefile.target
--- kvm-75.broken/qemu/Makefile.target	2008-09-07 18:38:33.000000000 +0200
+++ kvm-75/qemu/Makefile.target	2008-09-21 01:39:07.000000000 +0200
@@ -317,6 +317,8 @@
 
 ifdef CONFIG_DYNGEN_OP
 OPC_H = gen-op.h dyngen-opc.h op.h
+
+$(LIBOBJS): dyngen-opc.h
 endif
 
 libqemu.a: $(LIBOBJS)
--- kvm-80/qemu/qemu-kvm-powerpc.c~orig	2008-12-07 11:12:19.000000000 +0100
+++ kvm-80/qemu/qemu-kvm-powerpc.c	2008-12-08 16:49:46.000000000 +0100
@@ -31,7 +31,7 @@
 
 void cpu_reset(CPUState *env)
 {
-	memset(env->breakpoints, 0, sizeof(env->breakpoints));
+	memset(&env->breakpoints, 0, sizeof(env->breakpoints));
 	cpu_ppc_reset(env);
 }
 
