--- kvm-86/kvm/kernel/configure.orig	2009-05-20 15:59:56.000000000 +0200
+++ kvm-86/kvm/kernel/configure	2009-06-18 00:21:45.000000000 +0200
@@ -94,24 +94,6 @@ if [ -d "$kerneldir/include2" ]; then
     kernelsourcedir=${kerneldir%/*}/source
 fi
 
-if [ -n "$no_uname" -a "$want_module" ]; then
-    if [ -e "$kerneldir/.kernelrelease" ]; then
-        depmod_version=`cat "$kerneldir/.kernelrelease"`
-
-    elif [ -e "$kerneldir/include/config/kernel.release" ]; then
-            depmod_version=`cat "$kerneldir/include/config/kernel.release"`
-    elif [ -e "$kerneldir/.config" ]; then
-	depmod_version=$(awk '/Linux kernel version:/ { print $NF }' \
-			 "$kerneldir/.config")
-    else
-        echo
-        echo "Error: kernelversion not found"
-        echo "Please make sure your kernel is configured"
-        echo
-        exit 1
-    fi
-fi
-
 rm -f include/asm include-compat/asm
 mkdir -p include
 ln -sf asm-"$karch" include/asm
