diff -Nru kvm-12.orig/user/kvmctl.h kvm-12/user/kvmctl.h
--- kvm-12.orig/user/kvmctl.h	2007-02-02 11:53:37.583093500 +0000
+++ kvm-12/user/kvmctl.h	2007-02-02 11:54:08.673036500 +0000
@@ -6,6 +6,7 @@
 #define KVMCTL_H
 
 #define __user /* temporary, until installed via make headers_install */
+#include <asm-generic/errno.h>
 #include <linux/kvm.h>
 #include <stdint.h>
 
diff -Nru kvm-12.orig/qemu/Makefile kvm-12/qemu/Makefile
--- kvm-12.orig/qemu/Makefile	2007-02-02 11:53:37.719102000 +0000
+++ kvm-12/qemu/Makefile	2007-02-02 12:02:58.414143250 +0000
@@ -8,7 +8,7 @@
 .PHONY: all clean distclean dvi info install install-doc tar tarbin \
 	speed test test2 html dvi info
 
-CFLAGS+=-Wall -O2 -g -fno-strict-aliasing -I.
+CFLAGS+=-Wall -O2 -g -fno-strict-aliasing -I. -I $(KERNELDIR)/include
 ifdef CONFIG_DARWIN
 CFLAGS+= -mdynamic-no-pic
 endif
diff -Nru kvm-12.orig/qemu/disas.c kvm-12/qemu/disas.c
--- kvm-12.orig/qemu/disas.c	2007-02-02 11:53:37.783106000 +0000
+++ kvm-12/qemu/disas.c	2007-02-02 12:07:15.578215000 +0000
@@ -3,6 +3,7 @@
 #include "dis-asm.h"
 #include "elf.h"
 #include <errno.h>
+#include <asm-generic/errno.h>
 
 #include "cpu.h"
 #include "exec-all.h"
diff -Nru kvm-12.orig/qemu/slirp/slirp.h kvm-12/qemu/slirp/slirp.h
--- kvm-12.orig/qemu/slirp/slirp.h	2007-02-02 11:53:37.671099000 +0000
+++ kvm-12/qemu/slirp/slirp.h	2007-02-02 12:06:21.142813000 +0000
@@ -10,6 +10,7 @@
 #endif
 #include "config.h"
 #include "slirp_config.h"
+#include <asm-generic/errno.h>
 
 #ifdef _WIN32
 # include <inttypes.h>
diff -Nru kvm-12.orig/qemu/vl.h kvm-12/qemu/vl.h
--- kvm-12.orig/qemu/vl.h	2007-02-02 11:53:37.583093500 +0000
+++ kvm-12/qemu/vl.h	2007-02-02 12:04:10.846670000 +0000
@@ -37,6 +37,7 @@
 #include <unistd.h>
 #include <fcntl.h>
 #include <sys/stat.h>
+#include <asm-generic/errno.h>
 
 #ifndef O_LARGEFILE
 #define O_LARGEFILE 0
diff -Nru kvm-12.orig/qemu/Makefile.target kvm-12/qemu/Makefile.target
--- kvm-12.orig/qemu/Makefile.target	2007-02-02 11:53:37.771105250 +0000
+++ kvm-12/qemu/Makefile.target	2007-02-02 12:00:28.884798250 +0000
@@ -16,7 +16,7 @@
 endif
 TARGET_PATH=$(SRC_PATH)/target-$(TARGET_BASE_ARCH)
 VPATH=$(SRC_PATH):$(TARGET_PATH):$(SRC_PATH)/hw:$(SRC_PATH)/audio
-DEFINES=-I. -I.. -I$(TARGET_PATH) -I$(SRC_PATH)
+DEFINES=-I. -I.. -I$(TARGET_PATH) -I$(SRC_PATH) -I$(KERNELDIR)/include
 ifdef CONFIG_USER_ONLY
 VPATH+=:$(SRC_PATH)/linux-user
 DEFINES+=-I$(SRC_PATH)/linux-user -I$(SRC_PATH)/linux-user/$(TARGET_ARCH)
